.github/scripts/rename_and_update_mip.js

```diff
@@ -6,41 +6,44 @@ const MIPsFolderPath = path.join(__dirname, "..", "..", "MIPs");
 
 function getMIPFiles() {
   const files = fs.readdirSync(MIPsFolderPath);
-  return files.filter((file) => file.match(/^(mip-(\d+|x))\.md$/i));
+  return files.filter((file) => file.match(/^mip-x(-[\w-]+)?\.md$/i));
 }
 
-function getNextMIPNumber(mipFiles) {
-  const numberedMipFiles = mipFiles.filter((file) => file.match(/^mip-\d+\.md$/i));
+function getNextMIPNumber() {
+  const allFiles = fs.readdirSync(MIPsFolderPath);
+  const numberedMipFiles = allFiles.filter((file) => file.match(/^mip-\d+\.md$/i));
   const mipNumbers = numberedMipFiles.map((file) => parseInt(file.match(/^mip-(\d+)\.md$/i)[1], 10));
   return mipNumbers.length > 0 ? Math.max(...mipNumbers) + 1 : 1;
 }
 
-
-function renameAndUpdateMIPFile() {
+function renameAndUpdateMIPFiles() {
   const mipFiles = getMIPFiles();
 
   if (mipFiles.length === 0) {
     console.log("No MIP files found.");
     return;
   }
 
-  const nextMIPNumber = getNextMIPNumber(mipFiles);
+  mipFiles.forEach(() => {
+    const nextMIPNumber = getNextMIPNumber();
 
-  const oldFilePath = path.join(MIPsFolderPath, "mip-x.md");
-  const newFilePath = path.join(MIPsFolderPath, `mip-${nextMIPNumber}.md`);
+    const oldFilePath = path.join(MIPsFolderPath, "mip-x.md");
```

Comment:
I'm not fully understanding how this script is able to assign multiple mip file names if this loop is hardcoded to only target `mip-x.md`

Reply:
Thanks, I attempted to make some changes to match on files matching the pattern mip-x*.md and then apply the automatic numbering to them.